src/components/BookList.js

```diff
@@ -1,14 +1,20 @@
-import React from 'react';
+import React, { useEffect } from 'react';
 import { useSelector, useDispatch } from 'react-redux';
 import Book from './Book';
-import { removeBook } from '../redux/books/booksSlice';
+import { fetchBooks, removeBook } from '../redux/books/booksSlice';
 
 const BookList = () => {
-  const books = useSelector((state) => state.books);
+  const books = useSelector((state) => state.books.book);
   const dispatch = useDispatch();
 
+  useEffect(() => {
+    dispatch(fetchBooks());
+  }, [dispatch]);
+
   const handleDelete = (itemId) => {
-    dispatch(removeBook(itemId));
+    dispatch(removeBook(itemId)).then(() => {
+      dispatch(fetchBooks());
+    });
```

Comment:
[OPTIONAL]greate work 👍🏼  👍🏼  , but we don't need the second call to the api fetchBooks since remove and add book updates the state in store and useSelectore hook is listening to any changes in the store and updates you book list accordingly